contracts/security-token/BasicKYC.sol

```diff
@@ -6,72 +6,73 @@
  * based on that.
  */
 
-pragma solidity ^0.4.18;
+pragma solidity ^0.4.24;
 
 import "./KYCInterface.sol";
-import "zeppelin/contracts/ownership/Ownable.sol";
+import "zeppelin/contracts/ownership/rbac/RBAC.sol";
 
 /**
  * @author TokenMarket /  Ville Sundell <ville at tokenmarket.net>
  */
-contract BasicKYC is Ownable, KYCInterface {
-    /** @dev This mapping contains address which have completed the KYC: */
-    mapping (address => bool) public whitelist;
-    /** @dev This mapping contains signature hashes which have been already used: */
-    mapping (bytes32 => bool) public hashes;
+contract BasicKYC is RBAC, KYCInterface {
+  /** @dev This mapping contains signature hashes which have been already used: */
+  mapping (bytes32 => bool) public hashes;
+  /** @dev Mapping of all the flags for all the users: */
+  mapping (address => uint256) public flags;
 
-    /** @dev this event is emitted when address is whitelisted, including the nonce:*/
-    event Whitelisted(address who, bool status);
+  /** @dev These can be used from other contracts to avoid typos with roles: */
+  string public constant ROLE_SIGNER = "signer";
+  string public constant ROLE_SETTER = "setter";
 
-    /** @dev Simple contructor, mainly because of a Populus bug. */
-    function BasicKYC() Ownable() {
-      // This is here for our verification code only
-    }
+  /**
+   * @dev Interal function for setting the flags, and emmiting the event
+   * @param user Address of the user whose flags we would like to set
+   * @param newFlags Whole set of 256 flags
+   */
+  function writeFlags(address user, uint256 newFlags) internal {
+    flags[user] = newFlags;
```

Comment:
Also shouldn't new flags just be multiples of 2. Wondering when a mode `0b11`(binary representation of 3) will be needed? It should be either be `0b01` or `0b10`. In that case we need a check like:
`require(newFlags % 2 == 0)`

Reply:
Also understood from the test below that mode `0b11` is possible for contract creator. mode `0b10` should *not* even be allowed to set for other customers.

Reply:
Here are two things here:
1) It is important that all the flags can be set at once, so gas would be saved. Also it makes the contract simpler, the mask change would happen at the caller's side. Caller would just supply the ready-to-use bitmask

2) Flags do not have any specific meaning to the BasicKYC itself: this is important so the contract lifespan would be as long as possible. All flags are treated as equals